### PR TITLE
Fix std::out_of_range exception in command file parsing

### LINKED ISSUE #1099: exiv2 aborts with registerNs
**Describe the bug**
Another crash found along with #1097 with fuzzing
exiv2 got aborted with the following report:
```
terminate called after throwing an instance of 'std::out_of_range'
  what():  basic_string::substr: __pos (which is 18446744073709551615) > this->size() (which is 0)
Aborted (core dumped)
```

**To Reproduce**
./exiv2 -m [exiv2_mo_crash_1.txt](https://github.com/Exiv2/exiv2/files/4185793/exiv2_mo_crash_1.txt) mo [f1.txt](https://github.com/Exiv2/exiv2/files/4185806/f1.txt)
(Just download first exiv2_mo_crash_1.txt file. It seems the second f1.txt file does not matter.)

**Desktop (please complete the following information):**
Ubuntu 16.04, clang 4.0.0

**Additional context**
It seems `parseLine` is passing an empty string to `registerNs` at `exiv2.cpp:1477`
stack trace of the crash execution:
```
#0  0x00007ffff6f0f428 in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:54
#1  0x00007ffff6f1102a in __GI_abort () at abort.c:89
#2  0x00007ffff754984d in __gnu_cxx::__verbose_terminate_handler() () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#3  0x00007ffff75476b6 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#4  0x00007ffff7547701 in std::terminate() () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#5  0x00007ffff7547919 in __cxa_throw () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#6  0x00007ffff75703f7 in std::__throw_out_of_range_fmt(char const*, ...) () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#7  0x00007ffff75dbcf8 in std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::substr(unsigned long, unsigned long) const ()
   from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#8  0x00007ffff799c496 in Exiv2::XmpProperties::registerNs (ns="", prefix="Exiff\262.LfkInfo.GPSDa")
    at /home/cheong/subjects/exiv2-0.27.2-Source/src/properties.cpp:2508
#9  0x00000000004199b2 in (anonymous namespace)::parseLine (modifyCmd=..., line="reg Exiff\262.LfkInfo.GPSDa \"", num=1)
    at /home/cheong/subjects/exiv2-0.27.2-Source/src/exiv2.cpp:1477
#10 0x0000000000418a21 in (anonymous namespace)::parseCmdFiles (modifyCmds=std::vector of length 0, capacity 0,
    cmdFiles=std::vector of length 1, capacity 1 = {...}) at /home/cheong/subjects/exiv2-0.27.2-Source/src/exiv2.cpp:1303
#11 0x00000000004175ce in Params::getopt (this=0x65a790, argc=5, Argv=0x7fffffffe328) at /home/cheong/subjects/exiv2-0.27.2-Source/src/exiv2.cpp:1087
#12 0x0000000000412c16 in main (argc=5, argv=0x7fffffffe328) at /home/cheong/subjects/exiv2-0.27.2-Source/src/exiv2.cpp:139
```
Thanks

### PR DESCRIPTION
**Pull request by kevinbackhouse**
Fixes #1099

### CODE REVIEW

**Review by piponazo — DISMISSED** [review]
LGTM Thanks!

I also think that we could take a look to the `Exiv2::XmpProperties::registerNs` function and check if it makes sense to accept calls when the 1st parameter is empty. But that can be done separately.

**Comment by piponazo** [comment]
Ups, I just noticed that windows is not happy about this change.

**Comment by kevinbackhouse** [comment]
The new test is failing on Windows because the error message has a ".exe" in it:

```
exiv2.exe: Error parsing -m option arguments
```

Do we have an environment variable that expands to "exiv2.exe" on Windows? I tried `$exiv2` but that doesn't work because it expands to the full path to exiv2.

**Comment by piponazo** [comment]
In the file `tests/suite.conf` you can find the following:

```
[ENV]
exiv2_path: EXIV2_BINDIR
binary_extension: EXIV2_EXT
```

I know that we need to setup `EXIV2_EXT` on windows to run the python tests correctly. Maybe you can use the variable `binary_extension` in the python tests. Otherwise you could inspect only the first line of the error and ignore the second one.

File: tests/suite.conf [hunk 1]
```diff
@@ -28,6 +28,7 @@ kerInvalidTypeValue: invalid type in tiff structure
 kerNotAJpeg : This does not look like a JPEG image
 kerNoImageInInputData: Input data does not contain a valid image
 addition_overflow_message: Overflow in addition
+exiv2exe: exiv2${ENV:binary_extension}
```
Comment (D4N) [inline tests/suite.conf]: This is not really worth putting into the global variables, unless it is used in more places.

```suggestion
```

File: tests/bugfixes/github/test_issue_1099.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    https://github.com/Exiv2/exiv2/issues/1099
+
+    An empty value in the command file causes a std::out_of_range exception.
+    """
+    url = "https://github.com/Exiv2/exiv2/issues/1099"
+
+    filename1 = path("$data_path/issue_1099_poc.txt")
+    filename2 = path("$data_path/issue_1099_poc.bin")
+    commands = ["$exiv2 -m $filename1 mo $filename2"]
+    stdout = ["""Usage: $exiv2exe [ options ] [ action ] file ...
```
Comment (D4N) [inline tests/bugfixes/github/test_issue_1099.py]: I would suggest to instead override `compare_stdout` as follows:
``` Python
def compare_stdout(self, i, command, got_stdout, expected_stdout):
    pass
```
That will cancel the stdout comparison, which imho is not worth checking here as it is completely unrelated.

Unless of course we want to check the stdout here too?

**Review by D4N — DISMISSED** [review]
Looks good to me, I've added some comments concerning the tests.

Concerning the file `issue_1099_poc.bin`, is it really required? In #1099 the reporter stated that they are unsure whether it is even needed.

File: tests/suite.conf (hunk 1 above)
Comment (kevinbackhouse) [reply tests/suite.conf]: I can't figure out how to make it work any other way. I have tried putting `exiv2${ENV:binary_extension}` or `exiv2$binary_extension` in the test file, but it won't expand the variable when the `$` is in the middle of the word. Do you know how to do it?

**Comment by kevinbackhouse** [comment]
@D4N: no, the `issue_1099_poc.bin` is not required. I only included it because I thought it might look confusing if the test used some other random file that was already in the test suite. But I can remove it if you prefer.

File: tests/suite.conf (hunk 1 above)
Comment (D4N) [reply tests/suite.conf]: Yeah, `$ENV` doesn't get expanded in tests, only in the config file. However, now you no longer need this variable, so we can remove it altogether?

**Review by D4N — COMMENTED** [review]
> @D4N: no, the `issue_1099_poc.bin` is not required. I only included it because I thought it might look confusing if the test used some other random file that was already in the test suite. But I can remove it if you prefer.

Just load a random file and add a comment above the line saying: file $foo is just a placeholder, it is not read and not required.
This is not super important, but our git repo is already pretty huge and thus I'd like to avoid adding more big (and especially useless) files.

File: tests/suite.conf (hunk 1 above)
Comment (kevinbackhouse) [reply tests/suite.conf]: Right now, I am still using it in the `stderr` check. If I add a `compare_stderr` too, then I could remove the global variable.

**Review by piponazo — APPROVED** [review]

**Pull request merged**